src/urllib3/connection.py

```diff
@@ -184,7 +184,7 @@ def putrequest(self, method, url, *args, **kwargs):
 
     def putheader(self, header, *values):
         """"""
-        if SKIP_HEADER not in values:
+        if not any(isinstance(v, str) and v == SKIP_HEADER for v in values):
```

Comment:
Thoughts on using `type(v) is str` instead of `isinstance()`? Also can we use the `-bb` option as you've done in brotlicffi to ensure these error in the future?

Reply:
Why `type(v) is str`?

Reply:
As mentioned in the initial comment, I can't enable `-bb` because of brotlicffi (and other issues that I haven't investiaged yet).

I think `isinstance()` is the pythonic way to do this, and it performs just as well as `type() is str` (consistently less than 100nsec as mesured by timeit on my laptop).

Reply:
Was looking for a speed win in this often-called function :) It's all good though, let's merge this!